### PR TITLE
Update TestScheduler.cs

### PR DESCRIPTION
**Pull request by brah-mcdude**
fixed bug #5010

### CODE REVIEW

File: src/core/Akka.TestKit/TestScheduler.cs [hunk 1]
```diff
@@ -88,10 +88,14 @@ public void AdvanceTo(DateTimeOffset when)
         {
             var scheduledTime = _now.Add(initialDelay ?? delay).UtcTicks;
 
-            if (!_scheduledWork.TryGetValue(scheduledTime, out var tickItems))
+            ConcurrentQueue<ScheduledItem> tickItems;
+            while (!_scheduledWork.TryGetValue(scheduledTime, out tickItems))
```
Comment (Aaronontheweb) [inline src/core/Akka.TestKit/TestScheduler.cs:92]: so this is compare-and-swap basically @brah-mcdude ?

Comment (Aaronontheweb) [reply src/core/Akka.TestKit/TestScheduler.cs:92]: Ah ok, I think I get this now after reading the original issue on https://github.com/akkadotnet/akka.net/issues/5010

Comment (brah-mcdude) [reply src/core/Akka.TestKit/TestScheduler.cs:92]: tickItems is a queue of all the items that we want to schedule for a specific scheduledTime tick. 
We need to extract this queue from the dictionary of all _scheduledWork. 
If TryGetValue fails, then we need to create tickItems and add it to _scheduledWork. 
However, if by the time that we call TryAdd, someone else has already added a new tickItems at the scheduledTime tick entry, our call to TryAdd will fail.  In such a scenario - we need to start over.
In other words, we need to loop while TryGetValue fails.  
Actually, our thread's SECOND attempt to TryAdd MUST ALWAYS succeed. 
So, theoretically, we could re-write it as a 3 iterations for loop: (int i = 0; i < 3; i++) that will throw an exception if i reaches 3.
